docs/tutorials/gallery_adding_an_assistant.py

```diff
@@ -0,0 +1,177 @@
+
+    def _default_answer(self, prompt: str, sources: list[Source]) -> str:
+        """
+        "Compute" the response to a given prompt. In this case,
+        the "computation" is a default string.
+
+        Args:
+            prompt: Prompt to be answered.
+            sources: Sources to use when answering answer the prompt.
+
+        Returns:
+            Answer.
+        """
+        return (
+            f"This is a default answer. There were {len(sources)} sources."
+            ""
+            f"The prompt was"
+            f"{prompt}"
+        )
+
+
+# %%
+# ## The Explanation
+
+# %%
+# !!! tip
+#
+#     For organizational purposes, this tutorial is divided into steps, but you may wish to perform
+#     Step 2 during Step 1 for manual testing or debugging.
+
+
+# %%
+# ### Step 0: Import Necessary Modules
+
+# %%
+# First start by importing some necessary components. Ragna uses the `typing` library:
+
+# %%
+# ```python
+# from typing import Iterator
+# ```
+
+# %%
+# Your assistant will subclass the [`Assistant`][ragna.core.Assistant] abstract base class and
+# [`Source`][ragna.core.Source] will be used to hold the documents or files sent to the LLM.
+
+# %%
+# ```python
+# from ragna.core import Assistant, Source
+# ```
+
+# %%
+# ### Step 1: Write the Assistant
+
+# %%
+# The main thing to do is to implement the `answer` abstract method. The
+# [`answer`][ragna.core.Assistant.answer] method is where you put the logic to access your LLM.
+# This could call an API directly, call other member functions of your assistant that call an API,
+# or call a local LLM. Ragna is designed to give you that flexibility.
+
+# %%
+# ```python
+#     def answer(self, prompt: str, sources: list[Source]) -> Iterator[str]:
+#         """Answer a prompt given some sources.
+#
+#         Args:
+#             prompt: Prompt to be answered.
+#             sources: Sources to use when answering answer the prompt.
+#
+#         Returns:
+#             Answer.
+#         """
+#         yield self._default_answer(prompt, sources)
+#
+#     def _default_answer(self, prompt: str, sources: list[Source]) -> str:
+#         """
+#         "Compute" the response to a given prompt. In this case,
+#         the "computation" is a default string.
+#
+#         Args:
+#             prompt: Prompt to be answered.
+#             sources: Sources to use when answering answer the prompt.
+#
+#         Returns:
+#             Answer.
+#         """
+#         return (
+#             f"This is a default answer. There were {len(sources)} sources."
+#             ""
+#             f"The prompt was"
+#             f"{prompt}"
+#          )
+# ```
+
+
+# %%
+# !!! note
+#     While including the `_default_answer` method seems to obfuscate the code, it is meant
+#     to demonstrate that you can add multiple methods to your assistant that help compute
+#     the response. Examples include calling an external API or preparing the sources to be
+#     used by the model.
```

Comment:
I don't understand the reasoning. We can assume that someone who wants to write their own assistant, comes with enough programming knowledge to know that you can call a function from another function. Thus, there is no reason for us to obfuscate our code here. Let's keep it as simple as possible.